cache/ringbuffer.go

```diff
@@ -0,0 +1,89 @@
+package cache
+
+import clientv3 "go.etcd.io/etcd/client/v3"
+
+type entry struct {
+	rev   int64
+	event *clientv3.Event
+}
+
+type ringBuffer struct {
+	buf               []entry
+	head, size        int
```

Comment:
I think it's more comfortable to keep `head` and `tail` instead of `head` and `size`